error: invalid memory address or nil pointer dereference

```sh
ts=2024-08-01T14:29:05.600Z caller=ovs_exporter.go:50 level=info msg="Starting ovs_exporter" version="(version=, branch=, revision=444d9d9706127e76525edcc202d65ac4d14ba4f2)"
ts=2024-08-01T14:29:05.600Z caller=ovs_exporter.go:51 level=info msg="Build context" build_context="(go=go1.22.2, platform=linux/amd64, user=, date=, tags=unknown)"
ts=2024-08-01T14:29:05.613Z caller=tls_config.go:313 level=info msg="Listening on" address=[::]:9272
ts=2024-08-01T14:29:05.613Z caller=tls_config.go:316 level=info msg="TLS is disabled." http2=false address=[::]:9272
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x922f88]

goroutine 28 [running]:
github.com/vexxhost/ovs_exporter/collector.(*InterfaceCollector).collectAdminState(_, _, {{0xc0001207b0, 0x24}, 0xc000293280, 0x0, 0x0, 0x0, {0x0, 0x0, ...}, ...})
  /src/collector/interface.go:120 +0x88
github.com/vexxhost/ovs_exporter/collector.(*InterfaceCollector).Collect(0xc00031b1a0, 0xc00049c060)
  /src/collector/interface.go:99 +0x254
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
  /go/pkg/mod/github.com/prometheus/client_golang@v1.17.0/prometheus/registry.go:455 +0x105
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather in goroutine 24
  /go/pkg/mod/github.com/prometheus/client_golang@v1.17.0/prometheus/registry.go:547 +0xbab
```

--- a/.gitignore
+++ b/.gitignore
@@ -1,0 +1,25 @@
+# idea
+.idea
+
+# Binaries for programs and plugins
+*.exe
+*.exe~
+*.dll
+*.so
+*.dylib
+
+# Test binary, built with `go test -c`
+*.test
+
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
+
+# Dependency directories (remove the comment below to include it)
+# vendor/
+
+# Go workspace file
+go.work
+go.work.sum
+
+# env file
+.env

--- a/collector/interface.go
+++ b/collector/interface.go
@@ -200,13 +200,20 @@ func (c *InterfaceCollector) collectBfdFlapCount(ch chan<- prometheus.Metric, if
 }
 
 func (c *InterfaceCollector) collectStatistics(ch chan<- prometheus.Metric, iface ovsmodel.Interface) {
+	mac := "unknown"
+	if iface.MACInUse != nil {
+		mac = *iface.MACInUse
+	} else {
+		level.Warn(c.logger).Log("msg", "MAC address is nil for interface", "interface", iface.Name)
+	}
+
 	for stat, value := range iface.Statistics {
 		ch <- prometheus.MustNewConstMetric(
 			c.ifaceStatistics,
 			prometheus.CounterValue,
 			float64(value),
 			iface.Name,
-			*iface.MACInUse,
+			mac,
 			stat,
 		)
 	}

--- a/collector/interface.go
+++ b/collector/interface.go
@@ -108,8 +108,15 @@ func (c *InterfaceCollector) Collect(ch chan<- prometheus.Metric) {
 
 func (c *InterfaceCollector) collectAdminState(ch chan<- prometheus.Metric, iface ovsmodel.Interface) {
 	adminState := float64(0)
-	if *iface.AdminState == "up" {
+	if iface.AdminState != nil && *iface.AdminState == "up" {
 		adminState = 1
+	}
+
+	mac := "unknown"
+	if iface.MACInUse != nil {
+		mac = *iface.MACInUse
+	} else {
+		level.Warn(c.logger).Log("msg", "MAC address is nil for interface", "interface", iface.Name)
 	}
 
 	ch <- prometheus.MustNewConstMetric(
@@ -117,7 +124,7 @@ func (c *InterfaceCollector) collectAdminState(ch chan<- prometheus.Metric, ifac
 		prometheus.GaugeValue,
 		adminState,
 		iface.Name,
-		*iface.MACInUse,
+		mac,
 	)
 }
 

